README.md

```diff
@@ -270,9 +270,17 @@ Option | Description | Default
 ------------ | ------------- | -------------
 `whenAuthed` | The action to take if the user is authenticated. Either `AuthAction.RENDER` or `AuthAction.REDIRECT_TO_APP`. | `AuthAction.RENDER`
 `whenUnauthed` | The action to take if the user is *not* authenticated. Either `AuthAction.RENDER` or `AuthAction.REDIRECT_TO_LOGIN`. | `AuthAction.RENDER`
-`appPageURL` | The redirect destination URL when we should redirect to the app. Can be a string or a function that receives `{ ctx }` and returns a URL. | `config.appPageURL`
-`authPageURL` | The redirect destination URL when we should redirect to the login page. Can be a string or a function that receives `{ ctx }` and returns a URL. | `config.authPageURL`
+`appPageURL` | The redirect destination URL when we should redirect to the app. Can be a string or a function that receives `{ ctx, AuthUser }` and returns a URL as a string or an object (see below for schema). | `config.appPageURL`
+`authPageURL` | The redirect destination URL when we should redirect to the login page. Can be a string or a function that receives `{ ctx, AuthUser }` and returns a URL as a string or an object (see below for schema). | `config.authPageURL`
+
+When either `appPageURL` or `authPageURL` are set as an object literal or an object literal is returned from a function, the object must have the following schema:
```

Comment:
Was there a request to make this work with the existing param instead of adding a new one? I ask because in the case of passing an object the naming doesn't feel like it makes a ton of sense (since it isn't just the url at that point right?)

Reply:
This was in accordance with Kevin's original suggestion [here](https://github.com/gladly-team/next-firebase-auth/issues/187#issuecomment-854164651). After our discussion, the approach became more clear and the repo sets is up already with the overloading of the `authPageURL` and `appPageURL` as a function. Currently, it reads a little weird, but it could be different to make it more dry, something like `appPageURL.location`. It fit more within the current patterns here to allow someone to express the metadata of the url in an object than to add another property to the config.

Reply:
In the process of looking into a slight schema adjustment, I noticed the SSR component has a bug and there's a section of untested code that didn't catch it. I'll have a fix up for this shortly.

Reply:
I've made a slight update to the schema and added some test coverage to the SSR component. Let me know if you have any more questions or concerns about the approach here.

Reply:
Great find - no I'm good, you answered all my questions here.